doc-generator/src/main/scala/Main.scala

```diff
@@ -0,0 +1,135 @@
+import scala.sys.process._
+import better.files.Dsl._
+import com.codacy.plugins.api.results.Pattern
+import com.codacy.plugins.api.results.Result.Level
+import com.codacy.plugins.api.results.Tool
+import play.api.libs.json.Json
+import com.codacy.plugins.api.results.Pattern.Subcategory
+import com.codacy.plugins.api.results.Pattern.Description
+import com.codacy.plugins.api.results.Pattern.DescriptionText
+
+object Main extends App {
+
+  def toInputStream(s: String) =
+    new java.io.ByteArrayInputStream(s.getBytes(java.nio.charset.StandardCharsets.UTF_8.name))
+
+  val targetDir = pwd / "doc-generator" / "target"
+  val clangExtraDir = targetDir / "clang-tools-extra"
+
+  if (!(clangExtraDir).exists && clangExtraDir.nonEmpty) {
+    val pb = Process(
+      command = Seq("git", "clone", "--depth", "1", "https://github.com/llvm-mirror/clang-tools-extra.git"),
+      cwd = Some(targetDir.toJava)
+    )
+    require(pb.! == 0, "git clone failed")
+  } else {
+    val pb = Process(command = Seq("git", "pull"), cwd = Some(clangExtraDir.toJava))
+    require(pb.! == 0, "git pull failed")
+  }
+
+  def isNotTitle(s: String): Boolean = !s.startsWith("#")
+
+  def fixRepoLinks(s: String): String = {
```

Comment:
It would be nice to have a comment explaining why we need this

Reply:
Done :white